frontend/src/compliance_reporting/__tests__/ScheduleSummaryContainer.test.js

```diff
@@ -0,0 +1,225 @@
+            creditsOffsetA:true  
+         }
+      },
+      scheduleState:{
+      },
+      snapshot: {
+         summary:{
+            lines:[]
+         }
+      },
+      recomputeRequest: jest.fn(),
+      validating: true,
+      readOnly: true,
+      updateScheduleState: jest.fn(),
+
+   }
+  let component = renderer.create(
+    <BrowserRouter>
+      <Provider store={store}>
+        <ScheduleSummaryContainer {...props} />
+      </Provider>
+    </BrowserRouter>
+  )
+  const updatedProps = {
+   ...props,
+   readOnly: false
+  };
+  component = renderer.create(
+   <BrowserRouter>
+     <Provider store={store}>
+       <ScheduleSummaryContainer {...updatedProps} />
+     </Provider>
+   </BrowserRouter>
+ )
+  const tree = component.toJSON()
+  expect(tree).toMatchSnapshot()
+});
+test('ScheduleSummaryContainer should display with no snapshot', () => {
+   const props = {
+      complianceReport:{
+         summary:{
+            dieselClassDeferred:1,
+            creditsOffsetB:1,
+            dieselClassPreviouslyRetained:1,
+            dieselClassRetained:1,
+            gasolineClassDeferred:1,
+            gasolineClassObligation:1,
+            gasolineClassPreviouslyRetained:1,
+            gasolineClassRetained:1,
+            creditsOffset:1,
+            creditsOffsetA:true  
+         }
+      },
+      scheduleState:{
+      },
+      snapshot: {
+         summary:{
+            lines:[]
+         }
+      },
+      readOnly: true,
+      updateScheduleState: jest.fn(),
+      loadInitialState:jest.fn(),
+      recomputeRequest:jest.fn()
+
+   }
+  const component = renderer.create(
+    <BrowserRouter>
+      <Provider store={store}>
+        <ScheduleSummaryContainer {...props} />
+      </Provider>
+    </BrowserRouter>
+  )
+
+  const tree = component.toJSON()
+  expect(tree).toMatchSnapshot()
+});
+test('ScheduleSummaryContainer should display', () => {
+   const props = {
+      validationMessages:{},
+      scheduleState:{},
+      complianceReport:{
+         hasSnapshot: false,
+         summary:{
+            dieselClassDeferred:1,
+            creditsOffsetB:1,
+            dieselClassPreviouslyRetained:1,
+            dieselClassRetained:1,
+            gasolineClassDeferred:1,
+            gasolineClassObligation:1,
+            gasolineClassPreviouslyRetained:1,
+            gasolineClassRetained:1,
+            creditsOffset:1,
+            creditsOffsetA:true  
+         }
+      },
+      snapshot: {
+         summary:{
+            lines:[]
+         }
+      },
+      readOnly: false,
+      updateScheduleState: jest.fn(),
+      recomputeRequest:jest.fn()
+   }
+  const component = renderer.create(
+    <BrowserRouter>
+      <Provider store={store}>
+        <ScheduleSummaryContainer {...props} />
+      </Provider>
+    </BrowserRouter>
+  )
+
+  const tree = component.toJSON()
+  expect(tree).toMatchSnapshot()
+})
+
+test('ScheduleSummaryContainer should display loading', () => {
```

Comment:
You need to change the test names to describe what you are testing in each specific test.